extensions/BeatBlox/index.js

```diff
@@ -576,162 +801,119 @@
+                                    await playNoteCommonBeats.apply(this, [drumTrackName,1, k.contents,undefined,true]);
+                                }
+                                else{
+                                const noteToPlay = drumToMidiNote(k);
+                                const receivedNote = parseNote(noteToPlay);
+                                await playNoteCommonBeats.apply(this, [drumTrackName,1, receivedNote,undefined,true]);
+                                }
+                            }
+
+                        }, { args: [], timeout: I32_MAX });
+                        
+                    }),
+                    new Extension.Block('hitDrumsOverDuration','command','music','hit over %noteDurations drum sequence %mult%drums',['Quarter', ['Kick']], function(duration,drum){
+                        setupProcess(this);
+                        if (drum.contents.length === 0) throw Error(`cannot be empty`);
+                        // if(drum.contents.some(value => drumToMidiNote(value) === "")) throw Error(`cannot play non-drum value`);
+                        if(duration == '') throw Error(`duration cannot be empty`);
+                        const durationInBeats = durationToBeats(duration);
+                        const numberOfNotes = drum.contents.length;
+                        this.runAsyncFn(async () => {
+                            const trackName = this.receiver.id;
+                            const drumTrackName = trackName+"Drum";
+                            for(const k of drum.contents){
+                                if(k instanceof List){
+                                    await playNoteCommonBeats.apply(this, [drumTrackName,(durationInBeats/numberOfNotes), k.contents,undefined,true]);
+                                }
+                                else{
+                                const noteToPlay = drumToMidiNote(k);
+                                const receivedNote = parseNote(noteToPlay);
+                                await playNoteCommonBeats.apply(this, [drumTrackName,(durationInBeats/numberOfNotes), receivedNote,undefined,true]);
+                                }
+                            }
+                        }, { args: [], timeout: I32_MAX });
+                        
+                    }),
+                    new Extension.Block('soundMetaData', 'reporter', 'music', '%soundMetaData of sound %snd', ['duration', ''], function (option, sound) {
                         if (sound === '') throw Error(`sound cannot be empty`);
-                        if (this.receiver.sounds.contents.length){
-                            for (let i = 0; i< this.receiver.sounds.contents.length; i++){
-                                if (sound === this.receiver.sounds.contents[i].name){
+                        if (option === '') throw Error(`option value cannot be empty`);
+                        if (this.receiver.sounds.contents.length) {
```

Comment:
the if looks unnecessary. also the inside could be a foreach loop rather than an index-based loop